### DISCUSSION TITLE
7.6.1+ unexpected behaviour with fieldArrays

### DISCUSSION THREAD

**Discussion by terebentina** [react-hook-form/react-hook-form#5275] [General]
Hi, 
I am hitting an unexpected behaviour with 7.6.1 and up: 
I have a form with the initialValues set to `{ name: 'foo', arr: [] }`
In 7.6.0 using a useWatch for the entire form **before** using useFieldArray returns `{ name: 'foo', arr: [] }`.
In 7.6.1 under the same conditions the useWatch first returns `{ name: 'foo', arr: [] }` and then `{ name: 'foo' }` once useFieldArray is called (it's losing the empty array field).

Using useWatch **after** useFieldArray makes 7.6.1 behave like 7.6.0
I created this sandbox to exemplify what I mean: https://codesandbox.io/s/misty-monad-xqkji?file=/src/App.js

How to reproduce:
- Open the console and notice how 7.6.0 logs `{ name: 'foo', arr: [] }`
- Switch RHF to 7.6.1
- Notice how it logs 
  ```
  { name: 'foo', arr: [] }
  { name: 'foo' }
  ```
Is it possible to keep the default value in 7.6.1 as 7.6.0 does? I am thinking that maybe I am missing some config option....
In my case it's not possible to move the preview to after the form.

Thank you for any insight

**Comment by tparvi** [comment]
I have the exact same issue. This is bad because now I cannot trust the typescript type definitions. If I say my form has `foo: string[]` then with useWatch or even in validation that field is missing if it is empty.

Did you find any solution to this?

**Comment by bluebill1049** [comment]
hey @tparvi @terandrei sorry for the late reply, I have missed this question... https://codesandbox.io/s/wispy-butterfly-1ozx8 use the latest next version solved the problem for me. let me know if you want to confirm.

**Reply by tparvi** [reply]
In my case the problem was that I was using `useWatch` inside container component and then passing those values to child components as props. During first render empty array was missing as @terandrei also had noticed but during second render array was there.

I was able to fix this by simply calling `useWatch` inside the child component instead of calling `useWatch` inside container component and passing those values as props.

**Reply by bluebill1049** [reply]
child component will gets run first, i think that's the root cause.
